cdist/conf/type/__grub_config/manifest

```diff
@@ -0,0 +1,398 @@
+[ -f "$__object/parameter/GRUB_TERMINAL_OUTPUT" ] && GRUB_TERMINAL_OUTPUT=$(cat "$__object/parameter/GRUB_TERMINAL_OUTPUT")
+[ -f "$__object/parameter/GRUB_SERIAL_COMMAND" ] && GRUB_SERIAL_COMMAND=$(cat "$__object/parameter/GRUB_SERIAL_COMMAND")
+[ -f "$__object/parameter/GRUB_DISABLE_LINUX_UUID" ] && GRUB_DISABLE_LINUX_UUID=$(cat "$__object/parameter/GRUB_DISABLE_LINUX_UUID")
+[ -f "$__object/parameter/GRUB_DISABLE_RECOVERY" ] && GRUB_DISABLE_RECOVERY=$(cat "$__object/parameter/GRUB_DISABLE_RECOVERY")
+[ -f "$__object/parameter/GRUB_VIDEO_BACKEND" ] && GRUB_VIDEO_BACKEND=$(cat "$__object/parameter/GRUB_VIDEO_BACKEND")
+[ -f "$__object/parameter/GRUB_BACKGROUND" ] && GRUB_BACKGROUND=$(cat "$__object/parameter/GRUB_BACKGROUND")
+[ -f "$__object/parameter/GRUB_THEME" ] && GRUB_THEME=$(cat "$__object/parameter/GRUB_THEME")
+[ -f "$__object/parameter/GRUB_GFXPAYLOAD_LINUX" ] && GRUB_GFXPAYLOAD_LINUX=$(cat "$__object/parameter/GRUB_GFXPAYLOAD_LINUX")
+[ -f "$__object/parameter/GRUB_DISABLE_OS_PROBER" ] && GRUB_DISABLE_OS_PROBER=$(cat "$__object/parameter/GRUB_DISABLE_OS_PROBER")
+[ -f "$__object/parameter/GRUB_OS_PROBER_SKIP_LIST" ] && GRUB_OS_PROBER_SKIP_LIST=$(cat "$__object/parameter/GRUB_OS_PROBER_SKIP_LIST")
+[ -f "$__object/parameter/GRUB_DISABLE_SUBMENU" ] && GRUB_DISABLE_SUBMENU=$(cat "$__object/parameter/GRUB_DISABLE_SUBMENU")
+[ -f "$__object/parameter/GRUB_ENABLE_CRYPTODISK" ] && GRUB_ENABLE_CRYPTODISK=$(cat "$__object/parameter/GRUB_ENABLE_CRYPTODISK")
+[ -f "$__object/parameter/GRUB_PRELOAD_MODULES" ] && GRUB_PRELOAD_MODULES=$(cat "$__object/parameter/GRUB_PRELOAD_MODULES")
+[ -f "$__object/parameter/GRUB_RECORDFAIL_TIMEOUT" ] && GRUB_RECORDFAIL_TIMEOUT=$(cat "$__object/parameter/GRUB_RECORDFAIL_TIMEOUT")
+[ -f "$__object/parameter/GRUB_RECOVERY_TITLE" ] && GRUB_RECOVERY_TITLE=$(cat "$__object/parameter/GRUB_RECOVERY_TITLE")
+[ -f "$__object/parameter/GRUB_HIDDEN_TIMEOUT" ] && GRUB_HIDDEN_TIMEOUT=$(cat "$__object/parameter/GRUB_HIDDEN_TIMEOUT")
+[ -f "$__object/parameter/GRUB_HIDDEN_TIMEOUT_QUIET" ] && GRUB_HIDDEN_TIMEOUT_QUIET=$(cat "$__object/parameter/GRUB_HIDDEN_TIMEOUT_QUIET")
+[ -f "$__object/parameter/GRUB_HIDDEN_TIMEOUT_BUTTON" ] && GRUB_HIDDEN_TIMEOUT_BUTTON=$(cat "$__object/parameter/GRUB_HIDDEN_TIMEOUT_BUTTON")
+
+__file "$grub_default_path" \
+    --source "$__type/files/grub" \
+    --owner root --mode 0644
+
+# Default setttings 
+if [ -n "$GRUB_DEFAULT" ]; then
+__line GRUB_DEFAULT --file "$grub_default_path" \
+    --line GRUB_DEFAULT=\""$GRUB_DEFAULT"\" \
+    --state present 
+fi
+
+if [ -n "$GRUB_TIMEOUT" ]; then
+__line GRUB_TIMEOUT --file "$grub_default_path" \
+    --line GRUB_TIMEOUT=\""$GRUB_TIMEOUT"\" \
+    --state present
+fi
+
+if [ -n "$GRUB_DISTRIBUTOR" ]; then
+__line GRUB_DISTRIBUTOR --file "$grub_default_path" \
+    --line GRUB_DISTRIBUTOR=\""$GRUB_DISTRIBUTOR"\" \
+    --state present
+fi
+
```

Comment:
@Unterstrichmoepunterstrich Since there are many lines of the same form I would go with something like: 
<pre>
for var in GRUB_DEFAULT \
               GRUB_TIMEOUT \
               ....; do
    __line $var --file "$grub_default_path" --line $var=\""$$var...
done
</pre>
And you probably need dependency on __file for each __line, so this would be:
<pre>
for var in GRUB_DEFAULT \
               GRUB_TIMEOUT \
               ....; do
    eval "var_val=\$$var"  # get a value from variable which var is referencing
    require="__file/..."  __line $var --file "$grub_default_path" --line $var=\""${var_val}"\" --state present
done
</pre>
This way you have list of many vars, but only one line using require="__file..." __line...